CS/XPO/Readme.md

```diff
@@ -21,109 +21,95 @@ Check the original example description first for more information on the demonst
+                                return serviceProvider.GetRequiredService<IConnectionStringProvider>().GetConnectionString();
+                            };
+                        });
 		```
 
 	* WinForms (`YourSolutionName\YourSolutionName.Win\Startup.cs`)
 
-		```cs{10-11}
+		```cs{7-12}
 		//...
 		public class ApplicationBuilder : IDesignTimeApplicationFactory {
 			public static WinApplication BuildApplication(string connectionString) {
 				var builder = WinApplication.CreateBuilder();
 				builder.UseApplication<SequenceGeneratorWindowsFormsApplication>();
 				//...
-				builder.ObjectSpaceProviders
-					.AddSecuredXpo((application, options) => {
-						options.ConnectionString = connectionString;
-						IXpoDataStoreProvider dataStoreProvider = XPObjectSpaceProvider.GetDataStoreProvider(connectionString, null, true);
-						GenerateUserFriendlyId.Module.SequenceGenerator.Initialize(dataStoreProvider);
-					})
-		```    
+				builder.Services.AddScoped<SequenceGeneratorProvider>();
+                builder.Services.Configure<SequenceGeneratorOptions>(opt => {
+                    opt.GetConnectionString = (serviceProvider) => {
+                        return serviceProvider.GetRequiredService<IConnectionStringProvider>().GetConnectionString();
+                    };
+                });
+		```
 
 	For applications with the Middle Tier Security:
 
-	1. Create a custom service.
-
-		```cs
-		using DevExpress.ExpressApp.AspNetCore;
-		using DevExpress.ExpressApp.AspNetCore.WebApi;
-		using DevExpress.ExpressApp.Xpo;
-		using Microsoft.Extensions.Options;
-		
-		namespace SeqXPOMiddle.MiddleTier {
-		    public class WebApiApplicationSetup : IWebApiApplicationSetup {
-		        public void SetupApplication(AspNetCoreApplication application) {
-		            IOptions<DataServerSecurityOptions> options = application.ServiceProvider.GetRequiredService<IOptions<DataServerSecurityOptions>>();
-		            IXpoDataStoreProvider dataStoreProvider = XPObjectSpaceProvider.GetDataStoreProvider(options.Value.ConnectionString, null);
-		            GenerateUserFriendlyId.Module.SequenceGenerator.Initialize(dataStoreProvider);
-		        }
-		    }
-		}
-		```
-  
-  	2. Register this service in your `YourSolutionName.MiddleTier` project (`YourSolutionName.MiddleTier\Startup.cs`):
+	* Middle Tier project (`YourSolutionName.MiddleTier\Startup.cs`):
 
-		```cs
+		```cs{4-10}
 		public class Startup
 		//...
 		public void ConfigureServices(IServiceCollection services) {
-			services.AddSingleton<IWebApiApplicationSetup, WebApiApplicationSetup>();
+			services.AddScoped<SequenceGeneratorProvider>();
+            services.Configure<SequenceGeneratorOptions>(opt => {
+                opt.GetConnectionString = (serviceProvider) => {
+                    var options = serviceProvider.GetRequiredService<IOptions<DataServerSecurityOptions>>();
+                    return options.Value.ConnectionString;
+                };
+            });
 		```
 
```

Comment:
Can you update the following line below?

Line 150/136:
`1. As an alternative, you can implement much simpler solutions at the database level or by using the built-in `DistributedIdGeneratorHelper.Generate` method. Refer to the following article for more details: [Auto-Generate Unique Number Sequence](https://www.devexpress.com/Support/Center/p/T567184").`

Line 151/137:
Please change "In application with Security System..." to "In an application with the Security System..."

Line 164/150:
Please update this line to the actual document title:

`* [Auto-Generate Unique Number Sequence](https://www.devexpress.com/Support/Center/p/T567184)`

Reply:
Done